src/Surfnet/StepupSelfService/SelfServiceBundle/Mock/RemoteVetting/MockRemoteVetController.php

```diff
@@ -142,6 +168,59 @@ private function getResponseData(Request $request, SamlResponse $samlResponse)
      */
     private function getFullRequestUri(Request $request)
     {
-        return $request->getSchemeAndHttpHost() . $request->getBasePath() .$request->getPathInfo();
+        return $request->getSchemeAndHttpHost() . $request->getBasePath() . $request->getPathInfo();
+    }
+
+    /**
+     * @param string $data
+     * @return array
+     */
+    private function parseAttributes($data)
+    {
+        @json_decode($data);
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            throw new BadRequestHttpException(sprintf(
+                'Could not parse the attributes because no valid json was given %s',
+                $data
+            ));
+        }
+
+        $data = json_decode($data, true);
+
+        $result = [];
+        foreach ($data as $attr) {
+            if (!array_key_exists('name', $attr)) {
+                throw new BadRequestHttpException(sprintf(
+                    'Could not parse the attributes because no valid name was given %s',
+                    json_encode($data)
+                ));
+            }
+            if (!array_key_exists('value', $attr)) {
+                throw new BadRequestHttpException(sprintf(
+                    'Could not parse the attributes because no valid value was given %s',
+                    json_encode($data)
+                ));
+            }
+
+            if (!is_array($attr['value'])) {
+                throw new BadRequestHttpException(sprintf(
+                    'Could not parse the attributes because a value should be an array with strings %s',
```

Comment:
Consider allowing `string` and `string[]` as the saml spec also adheres to this. And what of numeric an boolean values for the 'value'?

Reply:
This is not an infrastructure layer so Saml spec seems irrelevant. Ans if look at the used Saml2 library the attributes in in an Assertion are `The attributes, as an associative array, indexed by attribute name`.